components/script/dom/bindings/htmlconstructor.rs

```diff
@@ -343,3 +352,109 @@ pub fn pop_current_element_queue() {
 pub fn push_new_element_queue() {
     ScriptThread::push_new_element_queue();
 }
+
+unsafe fn prepare_html_constructor_prototype(
+    cx: JSContext,
+    args: CallArgs,
+    get_proto_object: fn(JSContext, HandleObject, MutableHandleObject),
+) -> Result<(), Error> {
+    // Step 2 https://html.spec.whatwg.org/multipage/#htmlconstructor
+    // The custom element definition cannot use an element interface as its constructor
+    // The new_target might be a cross-compartment wrapper. Get the underlying object
+    // so we can do the spec's object-identity checks.
+    rooted!(in(*cx) let new_target = UnwrapObjectDynamic(args.new_target().to_object(), *cx, 1));
+    if new_target.is_null() {
+        return Err(Error::Type("new.target is null".to_owned()));
+    }
+    if args.callee() == new_target.get() {
+        return Err(Error::Type(
+            "new.target must not be the active function object".to_owned(),
+        ));
+    }
+
+    // Step 6
+    rooted!(in(*cx) let mut prototype = ptr::null_mut::<JSObject>());
+    {
+        rooted!(in(*cx) let mut proto_val = UndefinedValue());
+        let _ac = JSAutoRealm::new(*cx, new_target.get());
+        if !JS_GetProperty(
+            *cx,
+            new_target.handle(),
+            b"prototype\\0".as_ptr() as *const _,
+            proto_val.handle_mut(),
+        ) {
+            return Err(Error::Type("false".to_owned()));
```

Comment:
I am not sure what type of error is appropriate to return here. Previously it was a return false when it was within the monolithic function. For the sake of getting something "working" I made this placeholder error. Perhaps it is not suitable to throw an error here, and in that case I am not sure the best way of handling this.